### PR TITLE
Feature/top100

### PR DESCRIPTION
**Pull request by ddfczm**
Top 100 lots implementation, this covers:

* Visit tracking (designed to minimize db writes + support no downtime top 100)
* Task engine to manually run and schedule recurring tasks
* Top 100 calculation task
* Top 100 UI + UI cleanup
* Database prune task
* Task admin UI which lets you see tasks and run them manually
* Host to host communication (GluonHostPool), used for API => Task server communication
* Host tracking (supports host discovery for communication)
* Host status admin UI
* Admin UI updates and fixes. Most dependencies updated.

**Important information**
1. There are a bunch of db change scripts to run
0007_fso_lot_visits.sql
0008_fso_hosts.sql
0009_fso_tasks.sql

2. You need to update bower dependencies and re-dist the admin UI

3. Servers no longer have their own shared secret. Instead its one secret for the whole pool. Make sure the root secret is set (top level in config.json) and remove secret values from cities, lot city arrays.

4. You need to add a task service in config.json for tasks to work, e.g.

```"tasks":{
		"enabled": true,
		"call_sign": "callisto",
        "binding": "0.0.0.0:35100",
		"internal_host": "127.0.0.1:35",
		"public_host": "127.0.0.1:35",
		"certificate": "auth.east.ea.com.pfx",

		"schedule":[
			{
				"cron":"0 3 * * *", 
				"task":"prune_database", 
				"timeout":3600,
				"parameter":{
				}
			},
			{
				"cron":"0 4 * * *", 
				"task":"top100", 
				"timeout":3600,
				"shard_id": 1,
				"parameter":{
				}
			}
		]
	}```

### CODE REVIEW

File: TSOClient/FSO.Server/config.json [hunk 1]
```diff
@@ -35,9 +35,18 @@
...
 
 		"tuning":{
-			"top100":{
-				"bonus_per_visitor_hour": 5,
-				"bonus_per_property_top100_rank": 10
+			"bonus":{
+				"property_bonus":{
+					"per_unit": 20,
+					"overrides":{
+						"1": 1500,
```
Comment (RHY3756547) [inline TSOClient/FSO.Server/config.json]: Nice!

Comment (ddfczm) [inline TSOClient/FSO.Server/config.json]: Oops, that per unit was meant to be 10. I was testing the config was picking up changes :/.

**Comment by RHY3756547** [comment]
LGTM, that's a lot of changes!

**Comment by ddfczm** [comment]
ikr, as mature as things seem new features are always a can of worms.

**Pull request merged**